app/services/resource_compare_service.rb

```diff
@@ -0,0 +1,218 @@
+# frozen_string_literal: true
+
+# Compares two PDCMetadata::Resource objects and provides a hash with the `differences`
+#
+# If there are no differences between the two objects `identical?` returns `true` and `differences == {}`
+# If there are differences the `differences` hash has the following structure:
+#
+# ```
+#    :field_name = {action:, from:, to:, value: }     for single-value fields (e.g. description)
+#    :field_name = [{action:, from: to:, value: }]    for multi-value fields (e.g. keywords)
+# ```
+#
+# The `action` indicates whether a value changed, was added, or deleted.
+#
+# rubocop:disable Metrics/ClassLength
+class ResourceCompareService
+  attr_reader :differences
+
+  def initialize(before, after)
+    @before = before
+    @after = after
+    @differences = {}
+    compare_resources
+  end
+
+  def identical?
+    @differences == {}
+  end
+
+  private
+
+    def compare_resources
```

Comment:
I really liked how you had compare_value in creator.  I wonder if you added a compare_value to the Resource types if a large quantity of this comparison could be simplified.  I expected you to diff the compare_value in this routine when I saw it.

Reply:
I am still on the fence on what will be best for this in the long term. But certainly a good idea to keep in mind as we add more (complex) properties to resource.